src/test/java/jeryl/fyp/logic/commands/DeleteCommandTest.java

```diff
@@ -17,19 +18,24 @@
 import jeryl.fyp.model.ModelManager;
 import jeryl.fyp.model.UserPrefs;
 import jeryl.fyp.model.student.Student;
+import jeryl.fyp.model.student.StudentId;
 
 /**
  * Contains integration tests (interaction with the Model) and unit tests for
  * {@code DeleteCommand}.
  */
 public class DeleteCommandTest {
 
+    final StudentId STUDENT_ID = new StudentId("A0123456G");
+
+    final StudentId INVALID_STUDENT_ID = new StudentId("ABCDEFG");
+
     private Model model = new ModelManager(getTypicalFypManager(), new UserPrefs());
 
     @Test
     public void execute_validIndexUnfilteredList_success() {
         Student studentToDelete = model.getFilteredStudentList().get(INDEX_FIRST_STUDENT.getZeroBased());
-        DeleteCommand deleteCommand = new DeleteCommand(INDEX_FIRST_STUDENT);
+        DeleteCommand deleteCommand = new DeleteCommand(STUDENT_ID);
```

Comment:
This student ID is not among the list of typical students, have you tried what happened if this is the case?